pkg/kubelet/dockershim/network/hostport/hostport_manager_test.go

```diff
@@ -106,13 +136,18 @@ func TestOpenCloseHostports(t *testing.T) {
 		execer:      exec.New(),
 	}
 
+	// open all hostports defined in the test cases
 	for _, tc := range openPortCases {
 		mapping, err := manager.openHostports(tc.podPortMapping)
+		for hostport, socket := range mapping {
+			manager.hostPortMap[hostport] = socket
```

Comment:
Why is this needed here? This seems like something that should either happen in the actual manager code, or else not be needed.

If it's just for the counting below, then that's really only testing the behavior of the test program, not the behavior of the manager...

Reply:
this was actually a bug in the test.
The same loop to add the open sockets to the hostPortMap map does happen in the manager.Add() method, https://github.com/kubernetes/kubernetes/blob/12fc5e50b9394cc3fa498b24a96c894d96b3bff2/pkg/kubelet/dockershim/network/hostport/hostport_manager.go#L107-L114

We need this because later we test the `closeHostports()` method that iterates over that map https://github.com/kubernetes/kubernetes/blob/12fc5e50b9394cc3fa498b24a96c894d96b3bff2/pkg/kubelet/dockershim/network/hostport/hostport_manager.go#L310-L327
in 
```
	for _, tc := range closePortCases {
		err := manager.closeHostports(tc.portMappings)
```

if we don't populate the map the closeHostports will always succeed because it's map. 
I've also added the counting below as a safety check and to make it easier to debug